jOOR/src/main/java/org/joor/Reflect.java

```diff
@@ -806,4 +812,69 @@ else if (void.class == type) {
     }
 
     private static class NULL {}
+
+    static class JavaSupport {
+
+        static final boolean JAVA_7_OR_HIGHER = isJava7orHigher();
+
+        static final boolean JAVA_8_OR_HIGHER = isJava8orHigher();
+
+        private static boolean isJava7orHigher(){
+            try {
+                Class.forName("java.util.Objects");
+                return true;
+            } catch (ClassNotFoundException ignore){
+                return false;
+            }
+        }
+
+        private static boolean isJava8orHigher(){
+            try {
+                Class.forName("java.util.Optional");
+                return true;
+            } catch (ClassNotFoundException ignore){
+                return false;
+            }
+        }
+
+        static class MethodHandleSupport {
+
+            static final Constructor<MethodHandles.Lookup> cachedLookupConstructor;
+
+            static {
+                if (JAVA_7_OR_HIGHER) {
+                    try {
+                        cachedLookupConstructor = MethodHandles.Lookup.class.getDeclaredConstructor(Class.class);
+
+                        if (!cachedLookupConstructor.isAccessible()) {
+                            cachedLookupConstructor.setAccessible(true);
+                        }

```

Comment:
This block could use jOOR directly, no? We might even remove the compile-time dependency on JDK 7 dependency on `MethodHandles` this way